src/scripts/components/DateRangeDropdown.jsx

```diff
@@ -0,0 +1,189 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import { Dropdown, DropdownMenuItem, DropdownMenuDivider, DropdownMenuItemWild, Date, Button, ResourceRight } from '.';
+
+class DateRangeDropdown extends React.Component {
+  state = {
+    isApplyEnabled: false,
+    startDate: moment(this.props.startDate),
+    endDate: moment(this.props.endDate),
+    minDate: moment(this.props.minDate),
+    maxDate: moment(this.props.maxDate),
+    dropdownMenuItems: this.props.dropdownMenuItems || [],
+    activeKey: this.props.activeKey,
+  };
+
+  componentDidMount() {
+    this.setState({ isApplyEnabled: false, startDate: moment(this.props.startDate), endDate: moment(this.props.endDate) });
+  }
+
+  componentDidUpdate(prevProps, prevState) {
```

Comment:
Shouldn't we be checking prevProps here to confirm there was a change in the actual data before triggering a render? I know it's not a hard and fast rule, just wondering what your perspective is here.

Reply:
@juanfabrega I tested this component. We are getting same values in prevProps and prevState. I don't require this function now as I am managing component re rendering by it's props and state.